src/Engines/SolrEngine.php

```diff
@@ -286,44 +286,42 @@ protected function performSearch(Builder $builder, array $options = [])
      */
     protected function filters(Builder $builder)
     {
-        $collection = collect($builder->wheres);
-        $query = $collection->reduce([$this, 'buildFilter']);
-
-        return $query;
+        return collect($builder->wheres)->reduce([$this, 'buildFilter']);
     }
 
     /**
      * build an individual filter.
      *
-     * @param string $query The current query string to pass to fq
-     * @param array  $item
-     * @return string
+     * @param array  $carry The current query string to pass to fq
+     * @param array  $data
+     * @return array
      */
-    public function buildFilter($query, $item)
+    public function buildFilter(array $carry = null, array $data): array
     {
-        if (! empty($query)) {
-            // prepend the boolean from this query
-            $query .= " {$item['boolean']} ";
-        } else {
-            //initialize the query string
-            $query = '';
-        }
-        if ($item['field'] === 'nested') {
+        $carryItems = $carry['items'] ?? [];
+
+        if ($data['field'] === 'nested') {
             // handle the nested queries recursively
-            $query .= ' ('.collect($item['queries'])->reduce([$this, 'buildFilter']).') ';
-        } elseif (is_array($item['query'])) {
-            $query .= '(';
-            $query .= collect($item['query'])
-                ->map(function ($qString, $key) use ($item) {
-                    return "{$item['field']}:$qString";
-                })
-                ->filter()
-                ->implode(' OR ');
-            $query .= ')';
+            $nested = array_reduce($data['queries'], [$this, 'buildFilter']);
+            $query = $nested['query'];
+            $items = $nested['items'];
         } else {
-            $query .= "{$item['field']}:{$item['query']}";
+            $field = $data['field'];
+            $mode = $data['mode'];
+            $items = is_array($data['query']) ? $data['query'] : [$data['query']];
+            $start = count($carryItems);
+            $query = implode(' OR ', array_map(function ($index) use ($field, $mode) {
```

Comment:
Should this be using a static OR or should it be using the boolean passed in the data array?
```suggestion
            $boolean = $data['boolean'] ?? 'OR';
            $query = implode(" $boolean ", array_map(function ($index) use ($field, $mode) {
```
That being said, I may be totally off base. I originally built this over a year ago so my memory of it's inner workings are a little dated...

Reply:
This is the operator used between the different values for the current field. ie: you pass a list of values and anyone of them should match.

It replicates this part of the old code:
```
 collect($item['query'])
                ->map(function ($qString, $key) use ($item) {
                    return "{$item['field']}:$qString";
                })
                ->filter()
                ->implode(' OR ');
``` 

The given boolean is used at line 323 below.

Reply:
Aha! I knew I was missing something, thanks for the reply. I'll get this change merged in soon.